MapboxNavigation/CarPlayManager.swift

```diff
@@ -298,13 +298,28 @@ public class CarPlayManager: NSObject {
 @available(iOS 12.0, *)
 extension CarPlayManager: CPInterfaceControllerDelegate {
     public func templateDidAppear(_ template: CPTemplate, animated: Bool) {
+        guard interfaceController?.topTemplate == mainMapTemplate else { return }
         if template == interfaceController?.rootTemplate, let carPlayMapViewController = carWindow?.rootViewController as? CarPlayMapViewController {
             let mapView = carPlayMapViewController.mapView
             mapView.removeRoutes()
             mapView.removeWaypoints()
-            mapView.userTrackingMode = .followWithCourse
+            mapView.setUserTrackingMode(.followWithCourse, animated: true)
         }
     }
+    public func templateWillDisappear(_ template: CPTemplate, animated: Bool) {
+
+        let isCorrectType = type(of: template) == CPSearchTemplate.self || type(of: template) == CPMapTemplate.self
```

Comment:
FYI, I think this preserves the zooming-in animation when transitioning from the navigation map template to the main map template. I think it’s perfectly fine as long as there’s an initial frame in which the route line and waypoints are still visible; otherwise, it feels gratuitous.

Reply:
That's exactly what I'm seeing in my test runs.